addons/binding/org.openhab.binding.solaredge/src/main/java/org/openhab/binding/solaredge/handler/SolarEdgeAggregateDataPolling.java

```diff
@@ -0,0 +1,67 @@
+/**
+ * Copyright (c) 2010-2018 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.solaredge.handler;
+
+import org.openhab.binding.solaredge.internal.command.AggregateDataUpdate;
+import org.openhab.binding.solaredge.internal.command.AggregatePeriod;
+import org.openhab.binding.solaredge.internal.command.SolarEdgeCommand;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Polling worker class. This is responsible for periodic polling of sensor data.
+ *
+ * @author Alexander Friese - initial contribution
+ *
+ */
+public class SolarEdgeAggregateDataPolling implements Runnable {
+    /**
+     * Logger
+     */
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+
+    /**
+     * Handler for delegation to callbacks.
+     */
+    private final SolarEdgeHandler handler;
+
+    /**
+     * Constructor.
+     *
+     * @param handler handler which handles results of polling
+     */
+    public SolarEdgeAggregateDataPolling(SolarEdgeHandler handler) {
+        this.handler = handler;
+    }
+
+    /**
+     * Poll the SolarEdge Webservice one time per call.
+     */
+    @Override
+    public void run() {
+        if (handler.getWebInterface() != null) {
+            logger.debug("polling SolarEdge aggregate data {}", handler.getConfiguration());
+
+            SolarEdgeCommand adu_day = new AggregateDataUpdate(handler, AggregatePeriod.DAY);
+            SolarEdgeCommand adu_week = new AggregateDataUpdate(handler, AggregatePeriod.WEEK);
+            SolarEdgeCommand adu_month = new AggregateDataUpdate(handler, AggregatePeriod.MONTH);
+            SolarEdgeCommand adu_year = new AggregateDataUpdate(handler, AggregatePeriod.YEAR);
+
+            try {
+                handler.getWebInterface().executeCommand(adu_day);
+                handler.getWebInterface().executeCommand(adu_week);
+                handler.getWebInterface().executeCommand(adu_month);
+                handler.getWebInterface().executeCommand(adu_year);
+            } catch (Exception e) {
+                logger.warn("Caught Exception: {}", e.getMessage());
```

Comment:
The code executing there is dealing with web requests. It is most likely that there are failures from time to time. Of course it should be possible to view the stack trace at least in debug mode. So I changed it to this:

            } catch (Exception e) {
                if (logger.isDebugEnabled()) {
                    logger.debug("Caught Exception: {}", e.getMessage(), e);
                } else {
                    logger.warn("Caught Exception: {}", e.getMessage());
                }
            }

Of course I could completely remove the try catch but this would perhaps result in a complete failure when one request fails.

Reply:
Ok, I checked this again. I found the reason for this generic try-catch. This code is executed by the scheduler in a separate Thread. It seems that any exception which occours is SILENTLY caught and the periodic code execution is then stopped. I had the same behaviour in my other binding (NibeUplink). It is really hard to detect errors if the code execution stops without any hint. In that case there was indeed a coding error (NPE) but it took me hours to find it.

I do not know if this behaviour of the scheduler can be changed. It is provided by the 'BaseThingHandler' so it is essential part of openhab.
At least I will now move this try-catch to another location because we have this code block twice.